tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -3784,8 +3853,52 @@ <h2>Show similarity to selected datapoint</h2>
           }
         },
 
+        computeClosestCounterfactual: function(exInd, distances) {
+          // Distances are indexed by example ids
+          const modelInferenceValueStr = this.strWithModelName_(
+            inferenceValueStr,
+            this.nearestCounterfactualModelIndex
+          );
+          let closestDist = Number.POSITIVE_INFINITY;
+          let closest = -1;
+          for (let i = 0; i < this.visdata.length; i++) {
+            // Skip examples with the same inference class as the selected
+            // examples.
+            if (
+              this.visdata[exInd][modelInferenceValueStr] ==
+              this.visdata[i][modelInferenceValueStr]
+            ) {
+              continue;
+            }
+            let dist = distances[i];
+            if (dist < closestDist) {
+              closestDist = dist;
+              closest = i;
+            }
+          }
+          if (closest != -1) {
+            // Display the counterfactual in dive and example viewer.
+            this.comparedIndices = [closest];
+            this.counterfactualExampleAndInference = this.examplesAndInferences[
+              closest
+            ];
+            this.compareTitle = 'Counterfactual value(s)';
+          }
+        },
+
         findClosestCounterfactual_: function() {
           const selected = this.selected[0];
+          // Custom distance function can only be used when local.
+          // If using custom distance function, request distances and return.
+          if (this.local && this.customDistanceFunctionSet) {
+            this.requestDistanceWithCallback(
+              selected,
+              'computeClosestCounterfactual',
```

Comment:
I am super surprised that you cannot pass the reference to the function. Can you not tack on function to the Event object?

Reply:
This function sends request to the backend to compute distances on the python side. It is called by multiple functions from js side so the python side needs to know which callback function to call on the js side after the computation ends. Flow is js(request distance for x) -> python(compute distance) -> js(finalize x with distances). Is there a way to pass a js function reference back to python?

Reply:
Added TODO note here so that if there is a support in TensorBoard mode to provide custom distance functions for counterfactuals then we should update this function to reflect that.